test/test_gpusim.cpp

```diff
@@ -146,3 +148,17 @@ BOOST_AUTO_TEST_CASE(FoldFingerprint)
     BOOST_CHECK_EQUAL(answer.size(), 1);
     BOOST_CHECK_EQUAL(answer[0], 63);
 }
+
+BOOST_AUTO_TEST_CASE(getNextGPU)
+{
+    unsigned int gpucount = gpusim::get_gpu_count();
+    if(gpucount == 0) return;
+    // Make sure first go around gets every valid GPU
+    for(unsigned int i=0; i<gpucount; i++) {
+        BOOST_CHECK_EQUAL(i, gpusim::get_next_gpu());
+    }
+    // Make sure second go around loops through them all again
+    for(unsigned int i=0; i<gpucount; i++) {
+        BOOST_CHECK_EQUAL(i, gpusim::get_next_gpu());
+    }
+}
```

Comment:
hahahahaha this is your only test.

Reply:
Well when the test coverage was so good from a starting point, why add new tests? ;)